bridge/scripts/mint-script.ts

```diff
@@ -0,0 +1,286 @@
+// https://app.safe.global/home?safe=eth:0x4A2FbE06004e37dE6Fe7Da59a53D14a407Def0ed
+const EXISTING_SAFE_ADDRESS = ethers.utils.getAddress(
+    process.env.SAFE_ADDRESS!
+);
+
+async function mint() {
+    const program = new Command();
+    program
+        .name("ad-hoc mint script")
+        .description("Transfer using gnosis safe")
+        .version("0.0.1");
+
+    program
+        .command("mint")
+        .description("mint ( ad-hoc mint )")
+        .argument("<string>", "destination( address )")
+        .argument("<string>", "amount( to transfer )")
+        .action(async (address: string, amount: string) => {
+            console.log("address", address);
+            console.log("amount", amount);
+            await main(address, amount);
+        });
+
+    program.parseAsync();
+}
+
+mint();
+
+async function initalizeSafe(existingAddress = EXISTING_SAFE_ADDRESS) {
+    console.log("initializeSafe safe address", EXISTING_SAFE_ADDRESS);
+    safeAddress = existingAddress;
+    const ethAdapterOwner1 = new EthersAdapter({
+        ethers,
+        signerOrProvider: owner1Signer,
+    });
+
+    safeSdkOwner1 = await Safe.create({
+        ethAdapter: ethAdapterOwner1,
+        safeAddress,
+    });
+}
+
+async function proposeMintTransaction(
+    amount: string,
+    to: string,
+    gasPrice: number
+) {
+    const WNCG_CONTRACT_ADDRESS = ethers.utils.getAddress(
+        process.env.WNCG_CONTRACT_ADDRESS!
+    );
+    const mintAmount = ethers.utils.parseUnits(amount, 18);
+
+    const contract = new ethers.Contract(
+        WNCG_CONTRACT_ADDRESS,
+        ["function mint(address account, uint256 amount) public"],
+        owner1Signer
+    );
+
+    const data = contract.interface.encodeFunctionData("mint", [
+        to,
+        mintAmount,
+    ]);
+
+    const safeTransactionData: SafeTransactionDataPartial = {
+        to: WNCG_CONTRACT_ADDRESS,
+        value: "0",
+        data,
+        gasPrice,
+    };
+
+    // Create a Safe transaction with the provided parametersx
+    const safeTransaction: SafeTransaction =
+        await safeSdkOwner1.createTransaction({ safeTransactionData });
+
+    // Deterministic hash based on transaction parameters
+    const safeTxHash = await safeSdkOwner1.getTransactionHash(safeTransaction);
+
+    // Sign transaction to verify that the transaction is coming from owner 1
+    const senderSignature = await safeSdkOwner1.signTransactionHash(safeTxHash);
+
+    const owner1SignerAddress = await owner1Signer.getAddress();
+
+    console.log({
+        safeAddress,
+        safeTransactionData: safeTransaction.data,
+        safeTxHash,
+        senderAddress: await owner1Signer.getAddress(),
+        senderSignature: senderSignature.data,
+    });
+    await safeService.proposeTransaction({
+        safeAddress: ethers.utils.getAddress(safeAddress),
+        safeTransactionData: safeTransaction.data,
+        safeTxHash,
+        senderAddress: ethers.utils.getAddress(owner1SignerAddress),
```

Comment:
L144 `senderAddress` uses `senderAddress: await owner1Signer.getAddress()` L151 `senderAddress` uses `senderAddress: ethers.utils.getAddress(owner1SignerAddress)`.

also `safeAddress: ethers.utils.getAddress(safeAddress)` is different compared to the log.

I think it's better to use same format, even though it's insignificant, always better to have same value for actual code and LOG message.

Reply:
fixed!
set log for same value ( checkSummed address )